### ISSUE TITLE
Should DELETE /config/admin cleanup all leased tokens?

### ISSUE THREAD

**Issue by TJM** [jfrog/artifactory-secrets-plugin#87]
DELETE /config/admin could potentially leave behind a bunch of leased (generated) tokens. 

Should it be cleaning all of those up?

If you don't delete /config/admin, but instead DISABLE the plugin mount, vault will try to cleanup all the leased tokens.

I am wondering if we provide this endpoint, if that means that we need to code in the cleanup?

**Comment by alexhung** [comment]
On first read, this make senses. I'm trying to think through if there're other scenarios in which deleting any tokens in Artifactory is undesirable.

**Comment by TJM** [comment]
Right, I am not sure if it is even possible to do. I am not sure if we should even provide the endpoint. We should probably look at other plugins?

**Comment by TJM** [comment]
Maybe we only cleanup if the token has been rotated? Once it has been rotated, it is under vault's control, so it should be revoked on deletion. If the token has not been rotated, there is a small possibility, that they got a single admin token and used it for more than one thing (bad practice), or they could be planning to maintain the admin token using another system.
